web/packages/design/src/utils/useResizeObserver.ts

```diff
@@ -16,46 +16,78 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-import { RefObject, useCallback, useLayoutEffect } from 'react';
+import {
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  type RefCallback,
+} from 'react';
 
 /**
- * useResizeObserver sets up a ResizeObserver for ref and calls callback on each resize.
+ * useResizeObserver creates a ResizeObserver which fires a callback
+ * when the provided element is resized. The callback is called with the
+ * ResizeObserverEntry. The observer is disconnected when the element is
+ * unmounted.
  *
- * It does not fire if ref.current.contentRect.height is zero, to account for a special case in
- * Connect where tabs are hidden using `display: none;`.
+ * Returns a ref callback to attach to the element to be observed – the element
+ * must be a HTMLElement but may be conditionally null.
  *
- * Uses a layout effect underneath. If ref is conditionally rendered, set enabled to false when ref
- * is null.
+ * @example
+ * // Basic usage
+ * const Component = () => {
+ *   const handleResize = (entry: ResizeObserverEntry) => {
+ *     console.log({ entryContentRect: entry.contentRect });
+ *   };
+ *
+ *   const ref = useResizeObserver<HTMLDivElement>(handleResize);
+ *
+ *   return <div ref={ref}>This div is being observed</div>;
+ * };
  */
-export function useResizeObserver(
-  ref: RefObject<HTMLElement>,
+export const useResizeObserver = <T extends HTMLElement = HTMLElement>(
   callback: (entry: ResizeObserverEntry) => void,
-  { enabled = true }
-): void {
-  const effect = useCallback(() => {
-    if (!ref.current || !enabled) {
-      return;
-    }
+  { fireOnZeroHeight = true } = {}
```

Comment:
I think by default `useResizeObserver` should not call on zero height. The old version was added to the project specifically for that – we had some shared code that was used in Web UI and Connect and since it did not check for height being equal to 0, the callback would fire any time you switched tabs in Connect.

We should also keep the original comment explaining why zero height has significance in this context.

Reply:
gotcha, can update this default then